packages/design-system/src/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import { useCheckbox } from "@react-aria/checkbox";
+import { useToggleState } from "@react-stately/toggle";
+import { useFocusRing } from "@react-aria/focus";
+
+import { CheckboxProps } from "./Checkbox.types";
+import { StyledCheckbox } from "./Checkbox.styles";
+
+function Checkbox(props: CheckboxProps) {
+  const { children, isDisabled, isIndeterminate } = props;
+  const state = useToggleState(props);
+  const ref = React.useRef(null);
+  const { inputProps } = useCheckbox(props, state, ref);
+  const { focusProps, isFocusVisible } = useFocusRing();
+  const id = `ads-checkbox-${props.value}`;
+
+  return (
+    <StyledCheckbox
+      isChecked={state.isSelected}
+      isDisabled={isDisabled}
+      isFocusVisible={isFocusVisible}
+      isIndeterminate={isIndeterminate || false}
```

Comment:
let's add false as a default prop instead? combined with typescript we then won't run into this.